# 02_data_processed

Aligned and processed peaks for Deaf1 ChIP-seq.

All code is in the README file of the accompanying folder.

## Table of Contents


### 00_fastqc ###

Initial FastQC. 

### 01_align_callpeak ###

Code and output files for alignment with `bwa`, duplicate marking with `sambamba`, calling peaks with `macs2`. 

- **010_bwa**: Contains aligned reads from `bwa`. Files are on GEO. 

- **011_bwa_srt**: Contains sorted `bam` files (filtered and unfiltered).
Filtered files have duplicates marked by `samblaster` and filtered by `sambamba`. 

- **012_sambamba_samplestats**: Contains QC info about flags from bwa_srt. 
- `_flagstats` files: Quick tabulation of flags generated from sambamba. Quickly check if duplicate flags are still
present after filtering. 
- `_stats` files: Detailed flag stats generated from samtools stats. Used to plot figures via plot-bamstats

- **013_macs2**: Files generated by `macs2` peak caller. 

- **014_chipqc**: QC report for ChIP-seq data, generated by the R `ChIPQC` package

### 02_idrmeme ###

- **021_idr**: Contains merged high quality peaks using `IDR`
- **022_meme**: Code and results for MEME motif enrichment. 


## Dependencies

Reads in data from: [data/01_data_raw](../01_data_raw/). 

Used as input to: [notebooks/01_QC_GO/01_QC_GO.md](../../notebooks/01_QC_GO/)

